contrib/jmx-metrics/src/test/groovy/io/opentelemetry/contrib/jmxmetrics/OtlpIntegrationTests.groovy

```diff
@@ -25,13 +25,15 @@ import io.opentelemetry.proto.metrics.v1.DoubleHistogramDataPoint
 import io.opentelemetry.proto.metrics.v1.ResourceMetrics
 import org.testcontainers.Testcontainers
 import spock.lang.Requires
+import spock.lang.Retry
 import spock.lang.Timeout
 import spock.lang.Unroll
 
 @Requires({
     System.getProperty('ojc.integration.tests') == 'true'
 })
-@Timeout(60)
+@Timeout(90)
+@Retry
```

Comment:
Why the `@Retry`?  This makes me wonder if a bad/flaky test could be hiding among the retries...

(Same question applies a few more times below)

Reply:
You're right, this is hiding flake where only partial sets of expected metrics are received and the test fails [here](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/contrib/jmx-metrics/src/test/groovy/io/opentelemetry/contrib/jmxmetrics/target-systems/CassandraIntegrationTests.groovy#L68).  There is no "wait for desired metrics" feature in our mock collector at the moment.  I had wanted to add this functionality as part of addressing https://github.com/open-telemetry/opentelemetry-java-contrib/issues/24.